projects/kit/components/pager/pager.component.html

```diff
@@ -0,0 +1,33 @@
+<div
+    class="t-items"
+    [style.transform]="'translateX(' + left() + 'px)'"
+>
+    <div
+        *tuiRepeatTimes="let index of count"
+        #item
+        waIntersectionObserver
+        waIntersectionThreshold="0"
+        class="t-item"
+        [class._active]="index === activeItemIndex"
+        [class._invisible]="!visible.has(index)"
+        [class._last]="index === count - 1"
+        [class._pre-last]="index === count - 2"
+        [class._visible]="visible.has(index)"
+        (waIntersectionObservee)="onIntersection($event, index)"
```

Comment:
Why do we need intersection observer? We have all the data we need inside index setter. BTW, let's change it to `ngOnChange` because length/max can also change at the same time so we want to be sure we have all the latest values.

Reply:
fixed